compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -751,10 +751,13 @@ convertInterfaces env mc = interfaceDefs
 
     convertRequires :: [(GHC.TyCon, Maybe SourceLoc)] -> ConvertM (S.Set (Qualified TypeConName))
     convertRequires requires = S.fromList <$> sequence
-      [ withRange mloc $ convertInterfaceTyCon env handleIsNotInterface iface
+      [ guardSupportsInterfaceRequires $ withRange mloc $ convertInterfaceTyCon env handleIsNotInterface iface
```

Comment:
I think this way the error should point to the required interface instead of the entire declaration of the requiring interface.

```suggestion
      [ withRange mloc $ guardSupportsInterfaceRequires $ convertInterfaceTyCon env handleIsNotInterface iface
```